### PR TITLE
Move settings page

### PR DESCRIPTION
**Pull request by strangerkir**
Addresses [ECUR-60](https://inpsyde.atlassian.net/browse/ECUR-60).

Create a new settings page in `WooCommerce` settings with a separate tab.

Options intentionally saved in one WP option. This is needed so options saved with previous plugin versions would be preserved. In the future, it would be a good idea to keep options separately.

Additionally:
- require WordPress for dev in composer.json;
- added strict types declaration where missed;
- removed not used old exceptions classes;

### CODE REVIEW

File: src/AdminPages/AdminController.php [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+	{
+		$formData = $_POST[$this->fieldsCollectionName] ?? null;
+
+		if($formData === null){
+			return;
+		}
+
+		if(! $this->isNonceValid())
+		{
+			//todo: add admin notice here.
```
Comment (mmaymo) [inline src/AdminPages/AdminController.php:145]: Just to make sure, can this be done later?

**Review by mmaymo — APPROVED** [review]
Just have a doubt regarding a todo note. All seems good.

**Comment by strangerkir** [comment]
> Just have a doubt regarding a todo note. All seems good.

I agree it's better to do at once. For now, I just added call of `wp_nonce_ays()` function. It shows an error message and breaks execution. Something like `wp_die()`. 

I have no time to implement admin notices but the current solution seems to be better than just `todo`. What do you think?

File: src/AdminPages/AdminController.php (hunk 1 above)
Comment (strangerkir) [reply src/AdminPages/AdminController.php]: I guess, it's better to do now. For now, I just added `wp_nonce_ays()` to print the error message and exit.

**Review by mmaymo — APPROVED** [review]

**Pull request merged**